AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/Support/MockIdentityProvider.swift

```diff
@@ -11,41 +11,19 @@ import ClientRuntime
 
 struct MockIdentityProvider: CognitoUserPoolBehavior {
 
-    typealias InitiateAuthCallback = (
-        InitiateAuthInput,
-        (Result<InitiateAuthOutputResponse,
-         SdkError<InitiateAuthOutputError>>) -> Void
-    ) -> Void
-
-    typealias RespondToAuthChallengeCallback = (
-        RespondToAuthChallengeInput,
-        (Result<RespondToAuthChallengeOutputResponse,
-         SdkError<RespondToAuthChallengeOutputError>>) -> Void
-    ) -> Void
-
-    typealias SignUpCallback = (
-        SignUpInput,
-        (Result<SignUpOutputResponse,
-         SdkError<SignUpOutputError>>) -> Void
-    ) -> Void
-
-    typealias ConfirmSignUpCallback = (
-        ConfirmSignUpInput,
-        (Result<ConfirmSignUpOutputResponse,
-         SdkError<ConfirmSignUpOutputError>>) -> Void
-    ) -> Void
-
-    typealias GlobalSignOutCallback = (
-        GlobalSignOutInput,
-        (Result<GlobalSignOutOutputResponse,
-         SdkError<GlobalSignOutOutputError>>) -> Void
-    ) -> Void
-
-    typealias RevokeTokenCallback = (
-        RevokeTokenInput,
-        (Result<RevokeTokenOutputResponse,
-         SdkError<RevokeTokenOutputError>>) -> Void
-    ) -> Void
+    typealias InitiateAuthCallback = (InitiateAuthInput) throws
+    -> InitiateAuthOutputResponse
+
+    typealias RespondToAuthChallengeCallback = (RespondToAuthChallengeInput) throws
+    -> RespondToAuthChallengeOutputResponse
+
+    typealias SignUpCallback = (SignUpInput) throws -> SignUpOutputResponse
+
+    typealias ConfirmSignUpCallback = (ConfirmSignUpInput) throws -> ConfirmSignUpOutputResponse
+
+    typealias GlobalSignOutCallback = (GlobalSignOutInput) throws -> GlobalSignOutOutputResponse
+
+    typealias RevokeTokenCallback = (RevokeTokenInput) throws -> RevokeTokenOutputResponse
```

Comment:
Should these all be `async` to match the methods that conform to CognitoUserPoolBehavior?  Also, a nit, but should they still be called `...Callback`?  Perhaps we should call it `...Response` or something now?

Reply:
Updated with renamed properties